### ISSUE TITLE
AdversaryLossWeight_incongruence

### ISSUE THREAD

**Issue by giandos200** [Trusted-AI/AIF360#276]
https://github.com/Trusted-AI/AIF360/blob/a5dac73786265415138ed6e2f8c926d9ee968ee5/aif360/algorithms/inprocessing/adversarial_debiasing.py#L68 
Hi, I'm a PhD student from Politecnico di Bari, Italy;
I was reproducing this algorithm following the paper and the AIF360 implementation but not all is equal to the paper. At page 6 the authors says: 
"Without tweaking, this algorithm ran into issues with local minima, and the resulting models were often closer to
demographic parity than equality of odds. We implemented
a technique that helped: by increasing the hyperparameter α
in Eqn. 1 over time, the predictor had a much easier time
learning to deceive the adversary and therefore more strictly
enforce equality of odds. We set α =
√
t (where t is the step
counter), and to avoid divergence we set the predictor’s step
size to η ∝ 1/t"
I think the term self.adversary_loss_weight should be update in each iteration as equal to sqrt(epoch) (for epoch in range(self.num_epochs)).

**Comment by nrkarthikeyan** [comment]
@pronics2004 ^^^. If we agree that this is a genuine issue, may be @giandos200 - you can open a PR.

**Comment by giandos200** [comment]
okay, another issue is that the classifier and the adversary network are trained before the nested gradient training in the paper. I'll open a PR and implement a pretraining cycle if the debias=True.

**Comment by giandos200** [comment]
@pronics2004 @nrkarthikeyan i'm trying to pass the continuous integration test for the Pull Request but did not pass the test Continuous Integration / build-py (3.6/3.7/3.8) since the adversary test is done with TF1 but the code i've pushed is upgraded to TF2.
